### PR TITLE
Port losses to Keras Core

### PR DESCRIPTION
**Pull request by ianstenbit**
🚧 This is an experimental feature branch, more details soon. 🚧

### CODE REVIEW

File: keras_cv/losses/centernet_box_loss.py [hunk 1]
```diff
@@ -61,40 +60,43 @@ class CenterNetBoxLoss(keras.losses.Loss):
     def __init__(self, num_heading_bins, anchor_size, **kwargs):
         super().__init__(**kwargs)
         self.num_heading_bins = num_heading_bins
-        self.anchor_size = anchor_size
+        self.anchor_size = ops.convert_to_tensor(anchor_size)
```
Comment (mattdangerw) [inline keras_cv/losses/centernet_box_loss.py:63]: this is curious, why the need to convert? I have been trying to avoid passing tensor types as init args.

**Review by mattdangerw — APPROVED** [review]
This looks good to me! Just one little question

File: keras_cv/losses/focal_test.py [hunk 2]
```diff
@@ -68,13 +57,26 @@ def test_output_shape_from_logits(self):
...
+        y_true = rng.uniform(size=(2, 8, 10)).astype("float64")
+        y_logits = rng.uniform(low=-1000, high=1000, size=(2, 8, 10)).astype(
+            "float64"
+        )
+        y_pred = ops.cast(ops.sigmoid(y_logits), "float32")
 
         focal_loss_on_logits = FocalLoss(from_logits=True)
         focal_loss = FocalLoss()
 
+        # TODO(ianstenbit): This probably warrants some more investigation.
```
Comment (jbischof) [inline keras_cv/losses/focal_test.py:70]: Is it just the loss scale that differs or also the performance metrics?

File: keras_cv/losses/simclr_loss.py [hunk 3]
```diff
@@ -12,12 +12,19 @@
...
 
-import tensorflow as tf
-from tensorflow import keras
+from keras_cv.backend import keras
+from keras_cv.backend import ops
 
 LARGE_NUM = 1e9
 
 
+def l2_normalize(x, axis):
```
Comment (jbischof) [inline keras_cv/losses/simclr_loss.py:21]: Maybe we need this in Keras Core?

**Review by jbischof — APPROVED** [review]
Looks great, thank you!

File: keras_cv/losses/simclr_loss.py (hunk 3 above)
Comment (mattdangerw) [reply keras_cv/losses/simclr_loss.py:21]: It's an interesting question. The numpy way would be `numpy.linalg.norm`, but I don't think `linalg` is covered by tfnp at all.

File: keras_cv/losses/focal_test.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/losses/focal_test.py:70]: The performance metrics match when I train the model with different backends -- I think the root cause here is op fusion of softmax + xent, but I'm not certain yet (hence leaving the TODO)

File: keras_cv/losses/simclr_loss.py (hunk 3 above)
Comment (ianstenbit) [reply keras_cv/losses/simclr_loss.py:21]: Probably a reasonable thing to exist in Keras Core as either a util or a layer.
For now I figured it was a simple enough little thing to leave in-line here.

File: keras_cv/losses/centernet_box_loss.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/losses/centernet_box_loss.py:63]: Good catch -- I needed this earlier due to some PyTorch device weirdness, but it got fixed in Keras Core so this is no longer needed.
